feat: add cashboxes screen

mautematico opened the pull request.

mautematico committed:
feat: add cashboxes screen

feat: add cashboxes route
feat: add non-editable cashbox component
feat: cashboxes action required icon
feat: add cashOut denomination to cashboxes
feat: edit cashboxes values
feat: new server empty cashIn and reset cashOut actions
feat: reset cashboxes from UI
fix: cashbox border, cashbox font
fix: move cashbox styles to its own file
fix: use default table for cashboxes-table
fix: better import
fix: TODO: find a better way to display cashbox reset errors
fix: TODO for cashout
fix: move cashboxestable closer to parent
fix: WIP use EditableTable instead of fakatable
wip: move to editabletable
fix: WIP split cashbox into view + input components that can be used with formik
feat: rewrite cashbox component into view + fromik
feat: WIP use editableTable instead of hand made table
feat: WIP cashboxes editable table
feat: split cashbox
feat: Yup validation schema for cashboxes editable table
feat: split cashbox into view+formik
feat: WIP use editableTable instead of faketable
feat: use editableTable instead of fakeTable
fix: custom CashboxesTable not needed anymore

mautematico commented:
![Screenshot_2020-02-18 Lamassu Admin(3)](https://user-images.githubusercontent.com/1238928/74727452-ccf34d80-5206-11ea-937b-45fa4b9bbc42.png)
![Screenshot_2020-02-18 Lamassu Admin(2)](https://user-images.githubusercontent.com/1238928/74727454-cd8be400-5206-11ea-8faf-6e291525646a.png)
![Screenshot_2020-02-18 Lamassu Admin(1)](https://user-images.githubusercontent.com/1238928/74727455-ce247a80-5206-11ea-955b-3f2fc16f36c9.png)
![Screenshot_2020-02-18 Lamassu Admin](https://user-images.githubusercontent.com/1238928/74727456-ce247a80-5206-11ea-9b78-acb9f766e515.png)

mautematico commented:
![Screenshot_2020-02-27 Lamassu Admin(5)](https://user-images.githubusercontent.com/1238928/75435409-95785580-5918-11ea-9f75-ee54277089d2.png)

mautematico commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create the cashbox screen](https://trello.com/c/2EJJMU25/37-create-the-cashbox-screen)

RafaelTaranto commented on new-lamassu-admin/src/components/cashboxes-table/Table.js:
```diff
@@ -0,0 +1,199 @@
+import { makeStyles } from '@material-ui/core/styles'
```
Why not just use the default table? There's no need for virtualization here and its mostly copy paste. There should be a better way to do it.

mautematico replied:
Fixed. Will do the same for the cashout-table

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/cashbox/Cashbox.js:
```diff
@@ -0,0 +1,191 @@
+import TextInputFormik from '../base/TextInput'
+
+// TODO: Why is this not working?
+// import { tomato, zodiac } from '../../../styling/variables'
```
This doesn't work because it's already being imported.

If we need to import with `../` better to just import from `src/`, just like the previous imports

mautematico replied:
fixed

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/cashbox/Cashbox.js:
```diff
@@ -0,0 +1,191 @@
+  link: {
+    marginTop: spacer
+  }
+})
```
Create a file for this

mautematico replied:
done

RafaelTaranto commented on new-lamassu-admin/src/pages/maintenance/Cashboxes.js:
```diff
@@ -0,0 +1,157 @@
+          cashOutDenominations: (data.config.cashOutDenominations ?? {})[
+            m.deviceId
+          ],
+          overrides: { top: {}, bottom: {} }
```
If this is hardcoded why not integrate it to the component?

mautematico replied:
Not exactly hardcoded, but a safe default that will eventually be modified back and forth; If it's okay, I'll add a TODO here and try to come back with a better solution.

RafaelTaranto commented on new-lamassu-admin/src/pages/maintenance/Cashboxes.js:
```diff
@@ -0,0 +1,157 @@
+  const [onEmpty] = useMutation(EMPTY_CASHIN_BILLS, {
+    onError: ({ graphQLErrors, message }) => {
+      const errorMessage = graphQLErrors[0] ? graphQLErrors[0].message : message
+      alert(JSON.stringify(errorMessage))
```
Add a todo here and at the other alert, this should not be final.

mautematico committed:
fix: wip use flat config

mautematico committed:
feat: use the newest editableTable options

fix: new editableTable takes care of setEditing()
fix: new save() parameters in use

mautematico committed:
fix: editableTable needs id to be set and uniq

mautematico committed:
fix: simplify layout by using TitleSection

mautematico committed:
fix: reads denominations config on flat style
